src/test/java/integration/TestRequestManager.java

```diff
@@ -0,0 +1,425 @@
+        ws1.setAuthInfo("","");
+        dataMgr = new DataManager();
+        sesSto = dataMgr.getSessionStorage();
+        reqMgr = new RequestManager(dataMgr, ws1,
+                () -> Assert.fail("Error sending request"), (e, m) -> Assert.fail("Wrong status code")) {
+            @Override
+            public void finishSubscribeToProject(long id, File[] files) {
+
+            }
+
+            @Override
+            public void finishCreateProject(Project project) {
+
+            }
+
+            @Override
+            public void finishDeleteProject(Project project) {
+
+            }
+        };
+    }
+
+    @After
+    public void cleanup() throws ConnectException, InterruptedException {
+        Semaphore waiter = new Semaphore(0);
+        cleanupUser(user1ID, user1Pass, waiter);
+        cleanupUser(user2ID, user2Pass, waiter);
+        waiter.tryAcquire(2, 10, TimeUnit.SECONDS);
+    }
+
+    private void cleanupUser(String userID, String userPass, Semaphore waiter) {
+        logger.info("Cleaning up user " + userID);
+        WSManager wsMgr = new WSManager(new ConnectionConfig(SERVER_URL, false, 5));
+
+        wsMgr.sendRequest(new UserLoginRequest(userID, userPass).getRequest(response -> {
+            if (response.getStatus() == 200) {
+                String senderToken = ((UserLoginResponse) response.getData()).getToken();
+                wsMgr.setAuthInfo(userID, senderToken);
+
+                wsMgr.sendRequest(new UserDeleteRequest().getRequest(null, null));
+            }
+            waiter.release();
+        }, errHandler));
+    }
+
+    @Test
+    public void testRequestManager() throws java.net.ConnectException, InterruptedException {
+        registerUser(user1ID, user1FirstName, user1LastName, user1Email, user1Pass);
+        testLogin(user1ID, user1Pass);
+        testFetchPermissionConstants();
+        proj1 = testCreateProject(proj1.getName());
+        registerUser(user2ID, user2FirstName, user2LastName, user2Email, user2Pass);
+        testAddUserToProject(proj1.getProjectID(), proj1.getName(), user2ID, "admin", apiConstants);
+        testRemoveUserFromProject(proj1.getProjectID(), proj1.getName(), user2ID);
+        proj2 = testCreateProject(proj2.getName());
+        List<Long> projectIDs = new ArrayList<>();
+        projectIDs.add(proj1.getProjectID());
+        projectIDs.add(proj2.getProjectID());
+        testFetchAndSubscribeAll(projectIDs);
+        testUnsubscribeFromProject(proj1.getProjectID());
+        testFetchProjects();
+        testSubscribeToProject(proj1.getProjectID());
+        testAddUserToProject(proj2.getProjectID(), proj2.getName(), user2ID, "write", apiConstants);
+        testDeleteProject(proj1.getProjectID());
+        testLogout();
+        testLogin(user2ID, user2Pass);
+        testRemoveSelfFromProject(proj2.getProjectID());
+        testLogout();
+        testLogin(user1ID, user1Pass);
+        deleteProject(proj2.getProjectID());
```

Comment:
Can we split these tests into cohesive blocks? It's hard to read what does what.